### PR TITLE
feat: Replace old collection card on profile

### LINKED ISSUE #5032: Use new collection card on profile as we have on explorer
replace it with new collection card - same as on explore

image
image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #5032
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

/bsx/u/bXjrHpyJzSczJuuqmtN6nqCnd5b7KFa5LK1rBtF1tUHDYzp8N?tab=collection

image

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
otherwise lgtm

File: queries/rmrk/subsquid/collectionListByAccount.graphql [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+      id
+      metadata
+      name
+      price
+      burned
+      currentOwner
+      events(where: { interaction_eq: BUY }) {
+        meta
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply queries/rmrk/subsquid/collectionListByAccount.graphql]: It only needs the events list to calculate the stats of collections, as we have done in collection explorer. Now I have removed some other unnecessary parameters.

[the remark this answers is not in the document]
Comment (vikiival) [reply queries/rmrk/subsquid/collectionListByAccount.graphql]: So please open issue in rubick / snek with the data you need.

I would rather dedicate ~1-2 and guide you how to do that then doing this on frontend.

**Comment by Jarsen136** [comment]
> So please open issue in rubick / snek with the data you need.
> I would rather dedicate ~1-2 and guide you how to do that then doing this on frontend.

@vikiival 
Thanks. I would try to list all the tasks related. 

indexer side:
- snek: https://github.com/kodadot/snek/issues/142
- rubick: https://github.com/kodadot/rubick/issues/184

frontend side:
1. update `collectionListWithSearch.graphql` for collection search
2. update `collectionListByAccount.graphql` for collection in the user profile page
3. refactor `components/collection/CollectionCard.vue` and `components/collection/CollectionDetail.vue`

**Comment by vikiival** [comment]
Hey I am closing this issue, if you want to try to write the resolver I am happy to guide. 

I do not have currently space to write it by myself.

https://www.youtube.com/watch?v=7eQu_KS1TJk&t=7s

**Comment by roiLeo** [comment]
Hey! can this be reopen?

**Pull request closed without merging**